diff --git a/AppBundle/Model/AbstractShip.php b/AppBundle/Model/AbstractShip.php
index 8ecc1b0..366c6d5 100644
--- a/AppBundle/Model/AbstractShip.php
+++ b/AppBundle/Model/AbstractShip.php
@@ -10,7 +10,19 @@ abstract class AbstractShip
 
     abstract public function getDefense();
 
-    abstract public function getSingleAttackEffectiveness();
+    abstract protected function getWeaponPower();
+
+    abstract protected function doesShipUseDevastatingAttack();
 
     abstract public function doesShipUseEvasiveManeuvering();
+
+    public function getSingleAttackEffectiveness()
+    {
+        // devastating attack?
+        if ($this->doesShipUseDevastatingAttack()) {
+            return 5 * $this->getWeaponPower();
+        }
+
+        return $this->getWeaponPower();
+    }
 }
diff --git a/AppBundle/Model/JediShip.php b/AppBundle/Model/JediShip.php
index 853a4e6..5c85c78 100644
--- a/AppBundle/Model/JediShip.php
+++ b/AppBundle/Model/JediShip.php
@@ -24,15 +24,13 @@ class JediShip extends AbstractShip
         return 100;
     }
 
-    public function getSingleAttackEffectiveness()
+    protected function getWeaponPower()
     {
-        $weaponPower = 10;
-
-        // devastating attack?
-        if (rand(0, 10) == 10) {
-            return 5 * $weaponPower;
-        }
+        return 10;
+    }
 
-        return $weaponPower;
+    protected function doesShipUseDevastatingAttack()
+    {
+        return rand(0, 10) > 5;
     }
 }
diff --git a/AppBundle/Model/NormalShip.php b/AppBundle/Model/NormalShip.php
index d272f60..0baca6e 100644
--- a/AppBundle/Model/NormalShip.php
+++ b/AppBundle/Model/NormalShip.php
@@ -47,18 +47,18 @@ class NormalShip extends AbstractShip
         return rand(0, 10) > 8;
     }
 
-    public function getSingleAttackEffectiveness()
+    public function getType()
     {
-        // devastating attack?
-        if (rand(0, 10) == 10) {
-            return 5 * $this->weaponPower;
-        }
+        return 'normal';
+    }
 
-        return $this->weaponPower;
+    protected function doesShipUseDevastatingAttack()
+    {
+        return rand(0, 10) == 10;
     }
 
-    public function getType()
+    protected function getWeaponPower()
     {
-        return 'normal';
+        return $this->weaponPower;
     }
 }
