diff --git a/AppBundle/Model/JediShip.php b/AppBundle/Model/JediShip.php
new file mode 100644
index 0000000..dc9e30d
--- /dev/null
+++ b/AppBundle/Model/JediShip.php
@@ -0,0 +1,16 @@
+<?php
+
+namespace AppBundle\Model;
+
+class JediShip extends NormalShip
+{
+    public function doesShipUseEvasiveManeuvering()
+    {
+        return rand(0, 10) > 5;
+    }
+
+    public function getType()
+    {
+        return 'jedi';
+    }
+}
diff --git a/AppBundle/Service/ShipLoader.php b/AppBundle/Service/ShipLoader.php
index 108465b..06cdcc4 100644
--- a/AppBundle/Service/ShipLoader.php
+++ b/AppBundle/Service/ShipLoader.php
@@ -2,6 +2,7 @@
 
 namespace AppBundle\Service;
 
+use AppBundle\Model\JediShip;
 use AppBundle\Model\NormalShip;
 
 class ShipLoader
@@ -26,6 +27,11 @@ class ShipLoader
             $ships[] = $this->createShipFromData($shipData);
         }
 
+        $jediShip = new JediShip('Our cool Jedi ship');
+        $jediShip->setWeaponPower(10);
+        $jediShip->setDefense(100);
+        $ships[] = $jediShip;
+
         return $ships;
     }
 
@@ -45,7 +51,6 @@ class ShipLoader
     {
         $name = $shipData['name'];
         $ship = new NormalShip($name);
-
         $ship->setWeaponPower($shipData['weapon_power']);
         $ship->setDefense($shipData['defense']);
 
