diff --git a/AppBundle/Service/CacheableShipLoader.php b/AppBundle/Service/CacheableShipLoader.php
new file mode 100644
index 0000000..34c0646
--- /dev/null
+++ b/AppBundle/Service/CacheableShipLoader.php
@@ -0,0 +1,38 @@
+<?php
+
+namespace AppBundle\Service;
+
+use Doctrine\Common\Cache\Cache;
+
+class CacheableShipLoader
+{
+    /**
+     * @var ShipLoader
+     */
+    private $shipLoader;
+
+    /**
+     * @var Cache
+     */
+    private $cache;
+
+    public function __construct(ShipLoader $shipLoader, Cache $cache)
+    {
+        $this->shipLoader = $shipLoader;
+        $this->cache = $cache;
+    }
+
+    public function getShips()
+    {
+        $key = 'all_ships';
+
+        if ($this->cache->contains($key)) {
+            return $this->cache->fetch($key);
+        }
+
+        $ships = $this->shipLoader->getShips();
+        $this->cache->save($key, $ships);
+
+        return $ships;
+    }
+}
diff --git a/config/config.yml b/config/config.yml
index 007cbfa..91ef34d 100644
--- a/config/config.yml
+++ b/config/config.yml
@@ -21,6 +21,10 @@ services:
         class: AppBundle\Service\ShipLoader
         arguments: ['@app_pdo']
 
+    cacheable_ship_loader:
+        class: AppBundle\Service\CacheableShipLoader
+        arguments: ['@ship_loader', '@ship_cache']
+
     random_ship_selector:
         class: AppBundle\Service\RandomShipSelector
         arguments: ['@ship_loader']
