diff --git a/AppBundle/Controller/ShipController.php b/AppBundle/Controller/ShipController.php
index 1eb61c2..5cba885 100644
--- a/AppBundle/Controller/ShipController.php
+++ b/AppBundle/Controller/ShipController.php
@@ -16,7 +16,7 @@ class ShipController extends ContainerAware
      */
     public function homepage()
     {
-        $pdo = $this->getPDO();
+        $pdo = $this->container->get('app_pdo');
 
         $shipLoader = new ShipLoader($pdo);
         $ships = $shipLoader->getShips();
@@ -34,7 +34,7 @@ class ShipController extends ContainerAware
      */
     public function battleAction()
     {
-        $pdo = $this->getPDO();
+        $pdo = $this->container->get('app_pdo');
 
         $randomShipSelector = new RandomShipSelector(new ShipLoader($pdo));
 
@@ -50,19 +50,4 @@ class ShipController extends ContainerAware
 
         return new Response($html);
     }
-
-    /**
-     * @return \PDO
-     */
-    private function getPDO()
-    {
-        $pdo = new \PDO(
-            'mysql:host=localhost;dbname=oop',
-            'summer',
-            'camp'
-        );
-        $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
-
-        return $pdo;
-    }
 }
diff --git a/config/config.yml b/config/config.yml
index bc43099..b1f2c47 100644
--- a/config/config.yml
+++ b/config/config.yml
@@ -1,5 +1,10 @@
 services:
-    # todo - put any service definitions here
+    app_pdo:
+        class: PDO
+        arguments:
+            - 'mysql:host=localhost;dbname=oop'
+            - 'summer'
+            - 'camp'
 
 framework:
     secret: %secret%
