diff --git a/AppBundle/Controller/ShipController.php b/AppBundle/Controller/ShipController.php
index 7dc7ae4..1b11496 100644
--- a/AppBundle/Controller/ShipController.php
+++ b/AppBundle/Controller/ShipController.php
@@ -15,7 +15,14 @@ class ShipController extends ContainerAware
      */
     public function homepage()
     {
-        $shipLoader = new ShipLoader();
+        $pdo = new \PDO(
+            'mysql:host=localhost;dbname=ezsummer_oo',
+            'root',
+            ''
+        );
+        $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
+
+        $shipLoader = new ShipLoader($pdo);
         $ships = $shipLoader->getShips();
 
         $html = $this->container->get('twig')->render(
@@ -31,7 +38,14 @@ class ShipController extends ContainerAware
      */
     public function battleAction()
     {
-        $shipLoader = new ShipLoader();
+        $pdo = new \PDO(
+            'mysql:host=localhost;dbname=ezsummer_oo',
+            'root',
+            ''
+        );
+        $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
+
+        $shipLoader = new ShipLoader($pdo);
         $shipA = $shipLoader->getRandomShip();
 
         while ($shipA == ($shipB = $shipLoader->getRandomShip())) {
diff --git a/AppBundle/Service/ShipLoader.php b/AppBundle/Service/ShipLoader.php
index 796e9a4..108465b 100644
--- a/AppBundle/Service/ShipLoader.php
+++ b/AppBundle/Service/ShipLoader.php
@@ -8,6 +8,11 @@ class ShipLoader
 {
     private $pdo;
 
+    public function __construct(\PDO $pdo)
+    {
+        $this->pdo = $pdo;
+    }
+
     /**
      * @return NormalShip[]
      */
@@ -52,18 +57,6 @@ class ShipLoader
      */
     private function getPDO()
     {
-        if ($this->pdo === null) {
-            $pdo = new \PDO(
-                'mysql:host=localhost;dbname=ezsummer_oo',
-                'root',
-                ''
-            );
-
-            $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
-
-            $this->pdo = $pdo;
-        }
-
         return $this->pdo;
     }
 
