diff --git a/AppBundle/Controller/ShipController.php b/AppBundle/Controller/ShipController.php
index 67adff5..1eb61c2 100644
--- a/AppBundle/Controller/ShipController.php
+++ b/AppBundle/Controller/ShipController.php
@@ -16,12 +16,7 @@ class ShipController extends ContainerAware
      */
     public function homepage()
     {
-        $pdo = new \PDO(
-            'mysql:host=localhost;dbname=oop',
-            'summer',
-            'camp'
-        );
-        $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
+        $pdo = $this->getPDO();
 
         $shipLoader = new ShipLoader($pdo);
         $ships = $shipLoader->getShips();
@@ -39,12 +34,7 @@ class ShipController extends ContainerAware
      */
     public function battleAction()
     {
-        $pdo = new \PDO(
-            'mysql:host=localhost;dbname=oop',
-            'summer',
-            'camp'
-        );
-        $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
+        $pdo = $this->getPDO();
 
         $randomShipSelector = new RandomShipSelector(new ShipLoader($pdo));
 
@@ -60,4 +50,19 @@ class ShipController extends ContainerAware
 
         return new Response($html);
     }
+
+    /**
+     * @return \PDO
+     */
+    private function getPDO()
+    {
+        $pdo = new \PDO(
+            'mysql:host=localhost;dbname=oop',
+            'summer',
+            'camp'
+        );
+        $pdo->setAttribute(\PDO::ATTR_ERRMODE, \PDO::ERRMODE_EXCEPTION);
+
+        return $pdo;
+    }
 }
